TWLight/users/management/commands/survey_active_users.py

```diff
@@ -64,17 +76,31 @@ def handle(self, *args, **options):
                 # are 'active'
                 is_active=True,
             )
-            .order_by("last_login")
-        ):
-            # Send the email
+            .order_by("last_login")[:batch_size]
+        )
+
+        # No users qualify; exit
+        if not users.exists():
+            return
+
+        connection = get_connection(
+            backend="TWLight.emails.backends.mediawiki.EmailBackend"
+        )
+        connection.open()
+
+        # send the emails
+        for user in users:
             Survey.survey_active_user.send(
                 sender=self.__class__,
+                connection=connection,  # passing in the connection is what lets us handle these in bulk
                 user_email=user.email,
                 user_lang=user.userprofile.lang,
                 survey_id=options["survey_id"],
                 survey_langs=options["lang"],
             )
 
-            # Record that we sent the email so that we only send one.
-            user.userprofile.survey_email_sent = True
-            user.userprofile.save()
+        connection.close()
+
+        # Record that we sent the email so that we only send one.
+        # user.userprofile.survey_email_sent = True
```

Comment:
I commented this out for repeated local testing, but it's kind of bad anyway. If the djmail message created for this has a status of "draft", then the user wasn't actually sent the email. We're having to go check that manually for the cleanup command anyway.

I was thinking of just dropping the flag altogether and just switching to doing a lookup based on email address & subject, or adding a m2m user:message relationship, etc. I might recommend starting with a lookup and then only switching to a relationship if we really fork the middleware. The proxy model has some helper methods on it that you could use as inspiration for making this better.